corehq/util/pagination.py

```diff
@@ -182,6 +195,31 @@ def _get_event_handler(self):
         else:
             return ResumableIteratorEventHandler(self)
 
+    def retry(self, item_id, max_retry=3):
+        """Add document to be yielded at end of iteration
+
+        Iteration order of retry documents is undefined. All retry
+        documents will be yielded after the initial non-retry phase of
+        iteration has completed, and every retry document will be
+        yielded each time the iterator is stopped and resumed during the
+        retry phase. This method is relatively inefficient because it
+        forces the iteration state to be saved to couch. If you find
```

Comment:
Hmm, I see this always saves state in the "meta" couchdb, so this is still correct. I haven't thought through this completely, but the original design of saving to the db in which docs are being iterated was a feature. It had been possible perform the same iteration on multiple different databases and their state would be indepdendent. Now it seems like the iteration state would be shared unless we somehow incorporate the database name/backend into `iteration_name`.